src/Powercord/plugins/pc-emojiUtility/index.js

```diff
@@ -133,7 +168,286 @@ module.exports = class EmojiUtility extends Plugin {
+
+    const outer = this; // eslint-disable-line consistent-this
+
+    const MessageContextMenu = getModuleByDisplayName('MessageContextMenu');
+    inject('pc-emojiUtility-emojiContext', MessageContextMenu.prototype, 'render', function (args, res) { // eslint-disable-line func-names
+      const { target } = this.props; // eslint-disable-line no-invalid-this
+      if (target.classList.contains('emoji')) {
+        const matcher = target.src.match(outer.getEmojiUrlRegex());
+        if (matcher) {
+          let emoji = Object.values(emojiStore.getGuilds()).flatMap(g => g.emojis).find(e => e.id === matcher[1]);
+          if (emoji) {
+            emoji.fake = false;
+          } else {
+            emoji = outer.createFakeEmoji(matcher[1], target.alt.substring(1, target.alt.length - 1), target.src);
+          }
+
+          const getCloneableGuilds = () => {
+            const items = [];
+            const clonableGuilds = Object.values(getGuilds()).filter(guild => outer.hasPermission(guild.id, Permissions.MANAGE_EMOJIS));
+
+            const onClick = async (guild) => {
+              try {
+                await uploadEmoji(guild.id, await outer.getImageEncoded(emoji.url), emoji.name);
+
+                outer.sendBotMessage(outer.settings.get('useEmbeds')
+                  ? {
+                    type: 'rich',
+                    description: `Cloned emote ${outer.getFullEmoji(emoji)} to **${guild.name}**`,
+                    color: 65280
+                  }
+                  : `Cloned emote ${outer.getFullEmoji(emoji)} to **${guild.name}**`
+                );
+              } catch (error) {
+                console.error(error);
+
+                outer.sendBotMessage(outer.settings.get('useEmbeds')
+                  ? {
+                    type: 'rich',
+                    description: 'Failed to clone emote',
+                    color: 16711680,
+                    footer: {
+                      text: 'Check the console for more information'
+                    }
+                  }
+                  : 'Failed to clone emote, check the console for more information'
+                );
+              }
+            };
+
+            for (const guild of clonableGuilds) {
+              items.push({
+                type: 'button',
+                name: guild.name,
+                onClick: () => onClick(guild)
+              });
+            }
+
+            return items;
+          };
+
+          const getFeatures = () => {
+            const features = [];
+
+            features.push({
+              type: 'submenu',
+              name: 'Clone',
+              hint: 'to',
+              getItems: getCloneableGuilds
+            });
+
+            features.push({
+              type: 'button',
+              name: 'Save',
+              onClick: async () => {
+                if (!outer.settings.get('filePath')) {
+                  outer.sendBotMessage(outer.settings.get('useEmbeds')
```

Comment:
this goes for all of your sendBotMessage calls, it might be worthwhile to make a wrapper function that automatically handles string -> embed, e.g.
```js
const sendBotMessage = (content, embed = { color: 16711680 }) =>
  outer.sendBotMessage(
    outer.settings.get('useEmbeds');
      ? Object.assign({
          type: 'rich',
          description: content
        }, embed)
      : content;
  );
```